docker/dev/Dockerfile

```diff
@@ -52,7 +52,7 @@ USER craylabs
 
 RUN cd SmartSim && SMARTSIM_SUFFIX=dev python -m pip install .
 
-RUN python -m pip install smartsim[ml,dev]==0.5.0 jupyter jupyterlab matplotlib && \
+RUN python -m pip install smartsim[ml,dev]==0.5.1 jupyter jupyterlab matplotlib && \
```

Comment:
Could we pass the version using:

```
ARG SSVERSION 
... 
RUN python -m pip install smartsim[ml,dev]==${SSVERSION}
```

and avoid having to make this change in multiple places each time we release?

Could we also use this same key to generate ./doc/version.txt and avoid updating `doc/conf.py`?

Reply:
maybe better to do `ENV SMARTSIM_VERSION` instead to use the same source being used in [buildenv.py](https://github.com/CrayLabs/SmartSim/pull/386/files#diff-7945f9e10b608a1cc73eec4773459e082e858f3f7f37fed67c8a4fe6e0a7db5fR276)